# Netflix Movies & TV Shows Data Analysis

## Importing Libraries

In [1]:
import pandas as pd
import numpy as np

# Loading The Data

In [2]:
data = pd.read_csv("netflix_titles.csv")

In [3]:
df = pd.DataFrame(data)

In [4]:
df

,Unnamed: 0,show_id,type,title,director,cast,country,date_added,release_year,rating,duration,listed_in,description
0,0,s1,Movie,Dick Johnson Is Dead,Kirsten Johnson,NaN,United States,"September 25, 2021",2020,PG-13,90 min,Documentaries,"As her father nears the end of his life, filmm..."
1,1,s2,TV Show,Blood & Water,NaN,"Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...",South Africa,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, TV Dramas, TV Mysteries","After crossing paths at a party, a Cape Town t..."
2,2,s3,TV Show,Ganglands,Julien Leclercq,"Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...",NaN,"September 24, 2021",2021,TV-MA,1 Season,"Crime TV Shows, International TV Shows, TV Act...",To protect his family from a powerful drug lor...
3,3,s4,TV Show,Jailbirds New Orleans,NaN,NaN,NaN,"September 24, 2021",2021,TV-MA,1 Season,"Docuseries, Reality TV","Feuds, flirtations and toilet talk go down amo..."
4,4,s5,TV Show,Kota Factory,NaN,"Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...",India,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, Romantic TV Shows, TV ...",In a city of coaching centers known to train I...
...,...,...,...,...,...,...,...,...,...,...,...,...,...
8802,8802,s8803,Movie,Zodiac,David Fincher,"Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...",United States,"November 20, 2019",2007,R,158 min,"Cult Movies, Dramas, Thrillers","A political cartoonist, a crime reporter and a..."
8803,8803,s8804,TV Show,Zombie Dumb,NaN,NaN,NaN,"July 1, 2019",2018,TV-Y7,2 Seasons,"Kids' TV, Korean TV Shows, TV Comedies","While living alone in a spooky town, a young g..."
8804,8804,s8805,Movie,Zombieland,Ruben Fleischer,"Jesse Eisenberg, Woody Harrelson, Emma Stone, ...",United States,"November 1, 2019",2009,R,88 min,"Comedies, Horror Movies",Looking to survive in a world taken over by zo...
8805,8805,s8806,Movie,Zoom,Peter Hewitt,"Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...",United States,"January 11, 2020",2006,PG,88 min,"Children & Family Movies, Comedies","Dragged from civilian life, a former superhero..."


##

#   Initial Data Exploration

In [5]:
df.shape

(8807, 13)

In [6]:
df.info

<bound method DataFrame.info of       Unnamed: 0 show_id     type                  title         director  \
0              0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1              1      s2  TV Show          Blood & Water              NaN   
2              2      s3  TV Show              Ganglands  Julien Leclercq   
3              3      s4  TV Show  Jailbirds New Orleans              NaN   
4              4      s5  TV Show           Kota Factory              NaN   
...          ...     ...      ...                    ...              ...   
8802        8802   s8803    Movie                 Zodiac    David Fincher   
8803        8803   s8804  TV Show            Zombie Dumb              NaN   
8804        8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805        8805   s8806    Movie                   Zoom     Peter Hewitt   
8806        8806   s8807    Movie                 Zubaan      Mozez Singh   

                                           

In [7]:
df.head

<bound method NDFrame.head of       Unnamed: 0 show_id     type                  title         director  \
0              0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1              1      s2  TV Show          Blood & Water              NaN   
2              2      s3  TV Show              Ganglands  Julien Leclercq   
3              3      s4  TV Show  Jailbirds New Orleans              NaN   
4              4      s5  TV Show           Kota Factory              NaN   
...          ...     ...      ...                    ...              ...   
8802        8802   s8803    Movie                 Zodiac    David Fincher   
8803        8803   s8804  TV Show            Zombie Dumb              NaN   
8804        8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805        8805   s8806    Movie                   Zoom     Peter Hewitt   
8806        8806   s8807    Movie                 Zubaan      Mozez Singh   

                                             

In [8]:
df.tail

<bound method NDFrame.tail of       Unnamed: 0 show_id     type                  title         director  \
0              0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1              1      s2  TV Show          Blood & Water              NaN   
2              2      s3  TV Show              Ganglands  Julien Leclercq   
3              3      s4  TV Show  Jailbirds New Orleans              NaN   
4              4      s5  TV Show           Kota Factory              NaN   
...          ...     ...      ...                    ...              ...   
8802        8802   s8803    Movie                 Zodiac    David Fincher   
8803        8803   s8804  TV Show            Zombie Dumb              NaN   
8804        8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805        8805   s8806    Movie                   Zoom     Peter Hewitt   
8806        8806   s8807    Movie                 Zubaan      Mozez Singh   

                                             

In [9]:
df.columns

Index(['Unnamed: 0', 'show_id', 'type', 'title', 'director', 'cast', 'country',
       'date_added', 'release_year', 'rating', 'duration', 'listed_in',
       'description'],
      dtype='str')

In [10]:
df.dtypes

Unnamed: 0      int64
show_id           str
type              str
title             str
director          str
cast              str
country           str
date_added        str
release_year    int64
rating            str
duration          str
listed_in         str
description       str
dtype: object

In [11]:
df.isnull()

,Unnamed: 0,show_id,type,title,director,cast,country,date_added,release_year,rating,duration,listed_in,description
0,False,False,False,False,False,True,False,False,False,False,False,False,False
1,False,False,False,False,True,False,False,False,False,False,False,False,False
2,False,False,False,False,False,False,True,False,False,False,False,False,False
3,False,False,False,False,True,True,True,False,False,False,False,False,False
4,False,False,False,False,True,False,False,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...
8802,False,False,False,False,False,False,False,False,False,False,False,False,False
8803,False,False,False,False,True,True,True,False,False,False,False,False,False
8804,False,False,False,False,False,False,False,False,False,False,False,False,False
8805,False,False,False,False,False,False,False,False,False,False,False,False,False


In [12]:
df.isnull().sum()

Unnamed: 0         0
show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

In [13]:
df.duplicated()

0       False
1       False
2       False
3       False
4       False
        ...  
8802    False
8803    False
8804    False
8805    False
8806    False
Length: 8807, dtype: bool

In [14]:
df.describe()

,Unnamed: 0,release_year
count,8807.000000,8807.000000
mean,4403.000000,2014.180198
std,2542.506244,8.819312
min,0.000000,1925.000000
25%,2201.500000,2013.000000
50%,4403.000000,2017.000000
75%,6604.500000,2019.000000
max,8806.000000,2021.000000


# Data Cleaning

### Question 1: Which columns contain missing values in the dataset?



In [15]:
df.isnull().sum()

Unnamed: 0         0
show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

### Observation

The dataset contains missing values in six columns:

- director: 2,634 missing values
- cast: 825 missing values
- country: 831 missing values
- date_added: 10 missing values
- rating: 4 missing values
- duration: 3 missing values

The remaining columns do not contain any missing values.

## Question 2: Remove the duplicate rows from the dataset?

In [16]:
df.duplicated().sum()

np.int64(0)

### Observation:
There NO Duplicate rows in the Dataset

In [17]:
df = df.drop_duplicates()

In [18]:
df

,Unnamed: 0,show_id,type,title,director,cast,country,date_added,release_year,rating,duration,listed_in,description
0,0,s1,Movie,Dick Johnson Is Dead,Kirsten Johnson,NaN,United States,"September 25, 2021",2020,PG-13,90 min,Documentaries,"As her father nears the end of his life, filmm..."
1,1,s2,TV Show,Blood & Water,NaN,"Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...",South Africa,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, TV Dramas, TV Mysteries","After crossing paths at a party, a Cape Town t..."
2,2,s3,TV Show,Ganglands,Julien Leclercq,"Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...",NaN,"September 24, 2021",2021,TV-MA,1 Season,"Crime TV Shows, International TV Shows, TV Act...",To protect his family from a powerful drug lor...
3,3,s4,TV Show,Jailbirds New Orleans,NaN,NaN,NaN,"September 24, 2021",2021,TV-MA,1 Season,"Docuseries, Reality TV","Feuds, flirtations and toilet talk go down amo..."
4,4,s5,TV Show,Kota Factory,NaN,"Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...",India,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, Romantic TV Shows, TV ...",In a city of coaching centers known to train I...
...,...,...,...,...,...,...,...,...,...,...,...,...,...
8802,8802,s8803,Movie,Zodiac,David Fincher,"Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...",United States,"November 20, 2019",2007,R,158 min,"Cult Movies, Dramas, Thrillers","A political cartoonist, a crime reporter and a..."
8803,8803,s8804,TV Show,Zombie Dumb,NaN,NaN,NaN,"July 1, 2019",2018,TV-Y7,2 Seasons,"Kids' TV, Korean TV Shows, TV Comedies","While living alone in a spooky town, a young g..."
8804,8804,s8805,Movie,Zombieland,Ruben Fleischer,"Jesse Eisenberg, Woody Harrelson, Emma Stone, ...",United States,"November 1, 2019",2009,R,88 min,"Comedies, Horror Movies",Looking to survive in a world taken over by zo...
8805,8805,s8806,Movie,Zoom,Peter Hewitt,"Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...",United States,"January 11, 2020",2006,PG,88 min,"Children & Family Movies, Comedies","Dragged from civilian life, a former superhero..."


In [19]:
df.shape

(8807, 13)

## Question 3: Handle missing values in the dataset?

In [20]:
df.isnull().sum()

Unnamed: 0         0
show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

In [21]:
df['director'] = df['director'].fillna("Unknown")
df['cast'] = df['cast'].fillna("Unknown")
df['country'] = df['country'].fillna("Unknown")
df['date_added'] = df['date_added'].fillna("Unknown")
df['rating'] = df['rating'].fillna(df['rating'].mode()[0])
df['duration'] = df['duration'].fillna("Unknown")

df.isnull().sum()

### Observation:
The dataset contained missing values in columns such as `director`, `cast`, `country`, `date_added`, `rating`, and `duration`. These missing values were handled by replacing them with appropriate values. After cleaning, all columns contain zero missing values, making the dataset ready for further analysis.

#### Final Observation

- The dataset was checked for missing values and duplicates.
- Missing values were handled appropriately.
- Duplicate rows were checked and no duplicate records were found.
- Data types and formatting were verified.
- After cleaning, the dataset is ready for further Exploratory Data Analysis (EDA).

# Data Analysis

### 1.What is the distribution of Movies and TV Shows on Netflix?

In [22]:
df["type"].value_counts()

type
Movie      6131
TV Show    2676
Name: count, dtype: int64

### Observation

- The analysis shows the distribution of content types available on Netflix.
- Movies make up the majority of the content, while TV Shows represent a smaller portion.
- This indicates that Netflix has a larger collection of movies compared to TV Shows in this dataset.

### 2.Which countries produce the most Netflix content?

In [23]:
df["country"].value_counts().head(10)

country
United States     2818
India              972
Unknown            831
United Kingdom     419
Japan              245
South Korea        199
Canada             181
Spain              145
France             124
Mexico             110
Name: count, dtype: int64

### Observation

- The analysis shows the countries that contribute the highest number of Netflix titles.
- The United States produces the most Netflix content compared to other countries.
- Other countries also contribute a significant number of movies and TV shows, showing Netflix has a global content library.

### 3. Which ratings are most common on Netflix?

In [24]:
df["rating"].value_counts().head(10)

rating
TV-MA    3211
TV-14    2160
TV-PG     863
R         799
PG-13     490
TV-Y7     334
TV-Y      307
PG        287
TV-G      220
NR         80
Name: count, dtype: int64

### Observation

- The analysis shows the distribution of different content ratings available on Netflix.
- TV-MA is the most common rating, followed by TV-14 and TV-PG.
- This indicates that Netflix has a large amount of mature and teen-oriented content in its library.

### 4. Which year had the highest number of content releases?

In [25]:
df["release_year"].value_counts().head(10)

release_year
2018    1147
2017    1032
2019    1030
2020     953
2016     902
2021     592
2015     560
2014     352
2013     288
2012     237
Name: count, dtype: int64

### Observation

- The analysis shows the number of Netflix content releases across different years.
- The year 2018 had the highest number of content releases in the dataset.
- This indicates that Netflix significantly expanded its content library during this period.

### 5. How many Netflix titles are produced by each country?

In [26]:
df.groupby("country")["title"].count().head(10)

country
, France, Algeria                                       1
, South Korea                                           1
Argentina                                              56
Argentina, Brazil, France, Poland, Germany, Denmark     1
Argentina, Chile                                        2
Argentina, Chile, Peru                                  1
Argentina, France                                       1
Argentina, France, United States, Germany, Qatar        1
Argentina, Italy                                        1
Argentina, Spain                                        8
Name: title, dtype: int64

### Observation

- The analysis shows the number of Netflix titles produced by different countries.
- The result displays the count of titles for each country in the dataset.
- It helps us understand the contribution of different countries to Netflix's content library.

## Project Conclusion

- The Netflix dataset was successfully explored, cleaned, and analyzed using Pandas .
- Missing values and duplicate records were checked during the data cleaning process.
- The analysis provided insights about content types, countries producing Netflix content, ratings distribution, and yearly content releases.
- Movies are more common than TV Shows in the dataset, and Netflix contains content from many countries.
- The dataset analysis helps in understanding Netflix's content trends and can be used for further data analysis projects.